File: moderate/Reverse-Groups/Reverse Groups.java
import java.io.*;
public class Main {
    public static void main (String[] args) throws IOException {
        File file = new File(args[0]);
        BufferedReader buffer = new BufferedReader(new FileReader(file));
        String line;
        while ((line = buffer.readLine()) != null) {
            line = line.trim();
            // Process line of input Here
            invertElements(line);
        }
    }
    private static void invertElements(String line){
		if (line.length()>0 && line.indexOf(";")>=0){
			String[] numsStr = line.split(",");
			int leng = numsStr.length;
			String last = numsStr[leng-1];
			int k = Integer.valueOf(last.substring(last.indexOf(";")+1));
			numsStr[leng-1] = last.substring(0,last.indexOf(";"));
			
			
            String result = "";
            for(int i=0;i<leng/k;i++){
				result = result.concat(GetReversed(numsStr,i,k));
            }
			for(int i=k*(leng/k);i<leng;i++){
				result = result = result.concat(numsStr[i] + ",");
			}
            
            result = result.substring(0,result.length()-1);
			System.out.println(result);
		}
	}
		
	private static String GetReversed(String[] numsStr,int i, int k){
		int init = i*k;
		int fin = i*k+k-1;
        String result = "";
        for(int j=fin;j>=init;j--){
            result = result.concat(numsStr[j] + ",");
        }
        return result;
	}
}

File: moderate/Reverse-Groups/Reverse-Groups.cs
using System;
using System.IO;
using System.Collections.Generic;

class Program
{
    static void Main(string[] args)
    {
        using (StreamReader reader = File.OpenText(args[0]))
        while (!reader.EndOfStream)
        {
            string line = reader.ReadLine();
            if (null == line)
                continue;
            invertElements(line);
        }
    }
    
    private static void invertElements(string line){
		string[] Str = line.Split(';');
		int k = Convert.ToInt32(Str[1]);
    	string[] numsStr = Str[0].Split(',');
		int leng = numsStr.Length;
		string result = "";
        for(int i=0;i<leng/k;i++){
			result += GetReversed(numsStr,i,k);
        }
		for(int i=k*(leng/k);i<leng;i++){
			result += (numsStr[i] + ",");
		}
        Console.WriteLine(result.Substring(0,result.Length-1));
	}

		
	private static string GetReversed(string[] numsStr,int i, int k){
		int init = i*k;
		int fin = i*k+k-1;
        string result = "";
        for(int j=fin;j>=init;j--){
            result += numsStr[j] + ",";
        }
        return result;
	}
}
